### ISSUE TITLE
update action generates migrations that are potentially unapplicable on top for existing migrations

### ISSUE THREAD

**Issue by gvlasov** [bizley/yii2-migration#30]
I'm on mysql 5.7 and Yii2 2.0.15.1

Here is the part of my Yii config related to this extension:
```
        'controllerMap' => [
            'migration-creator' => [
                'class' => 'bizley\migration\controllers\MigrationController',
                'generalSchema' => 0
            ],
        ]
```

My problem is that when I create a "create table" migration and then try to create an "update table" migration for the same table (after making sure the "create table" migration is applied), the update migration would contain alter statements that are already in the create migration. The most disturbing part is, in case the create migration has a primary key declaration, that update migration would be unapplicable, because it tires to create an additional primary key.

Here is the minimal example to reproduce the issue:

Here is a create migration generated with `yii migration-creator/create cache`:

```php
<?php

use yii\db\Migration;

class m180630_235851_create_table_cache extends Migration
{
    public function up()
    {
        $tableOptions = null;
        if ($this->db->driverName === 'mysql') {
            $tableOptions = 'CHARACTER SET utf8 COLLATE utf8_unicode_ci ENGINE=InnoDB';
        }

        $this->createTable('{{%cache}}', [
            'id' => $this->char(128)->notNull()->append('PRIMARY KEY'),
            'expire' => $this->integer(11),
            'data' => $this->binary(),
        ], $tableOptions);

    }

    public function down()
    {
        $this->dropTable('{{%cache}}');
    }
}
```

Here is how my database looks when I load nothing but that migration with `yii migrate/up`:

![image](https://user-images.githubusercontent.com/4467948/42129997-d2aa27ce-7cee-11e8-8568-b8bb0a2793c0.png)

Then, here is an update migration generated with `yii migration-creator/update cache`:

```php
<?php

use yii\db\Migration;

class m180701_000023_update_table_cache extends Migration
{
    public function up()
    {
        $this->alterColumn('{{%cache}}', 'id', $this->char(128)->notNull()->append('PRIMARY KEY'));
        $this->alterColumn('{{%cache}}', 'expire', $this->integer(11));
    }

    public function down()
    {
        echo "m180701_000023_update_table_cache cannot be reverted.\n";
        return false;
    }
}
```

When I apply it with `yii migration/up` after generating it, here is what I get:

```
$ bin/docker/yii migrate/up
Yii Migration Tool (based on Yii v2.0.15.1)

Total 1 new migration to be applied:
        m180701_002353_update_table_cache

Apply the above migration? (yes|no) [no]:yes
*** applying m180701_002353_update_table_cache
    > alter column id in table {{%cache}} to char(128) NOT NULL PRIMARY KEY ...Error: SQLSTATE[42000]: Syntax error or access violation: 1068 Multiple primary key defined
The SQL being executed was: ALTER TABLE `cache` CHANGE `id` `id` char(128) NOT NULL PRIMARY KEY
```

**Comment by gvlasov** [comment]
I forgot to add that I expected the extension to understand that those alter statements in the update migration should not be generated, because they just restate what already is in the schema, and not generate those.

BTW thanks for your work, this is a great extension.

**Comment by bizley** [comment]
Thanks for your feedback. Let me confirm this first and I'll get back to you.

**Comment by bizley** [comment]
Fixed in [3.0.1](https://github.com/bizley/yii2-migration/releases/tag/3.0.1)/[2.3.1](https://github.com/bizley/yii2-migration/releases/tag/2.3.1)

Thank you!
